### PR TITLE
coqPackages.lemma-overloading: init at 8.12

### PR DESCRIPTION
**Pull request by pi8027**
Add new Coq package [lemma-overloading](https://github.com/coq-community/lemma-overloading)

@CohenCyril 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by CohenCyril** [comment]
This is a good first draft. Now you need to try and build each version to get the hash.

**Comment by pi8027** [comment]
I noticed that I should remove the dependency on hierarchy-builder from the existing released versions of lemma-overloading. How should I do that?

**Comment by pi8027** [comment]
I guess I should rely on the dependency from MC to HB and transitivity.

**Comment by CohenCyril** [comment]
> I guess I should rely on the dependency from MC to HB and transitivity.

Indeed so I guess it has already been taken care of by the lack of dependency from MC < 2

File: pkgs/development/coq-modules/lemma-overloading/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "8.8.0".sha256 = "sha256-Iq3KfESMnZF8hhGKuvZHx+hAMEaoCP7MhhQEI6xfoO8=";
+  };
+  releaseRev = v: "v${v}";
+
+  propagatedBuildInputs = [ mathcomp-ssreflect ];
+
+  meta = {
+    description = "Libraries demonstrating design patterns for programming and proving with canonical structures in Coq";
+    maintainers = with lib.maintainers; [ cohencyril ];
+  };
```
Comment (cu1ch3n) [inline pkgs/development/coq-modules/lemma-overloading/default.nix:30]: Is the license missed?

Comment (pi8027) [reply pkgs/development/coq-modules/lemma-overloading/default.nix:31]: Fixed.

**Review by vbgl — APPROVED** [review]

**Comment by cu1ch3n** [comment]
To pass CI checks:

> Please go to the Nixpkgs root directory, run `nix-shell`, then:
nixfmt 'pkgs/development/coq-modules/lemma-overloading/default.nix'

**Comment by cu1ch3n** [comment]
@vbgl sorry but why the fmt is not needed?

**Pull request merged**